### PR TITLE
Bump containernetworking

### PR DESCRIPTION
**Pull request by jcaamano**
Bump containernetworking to latest version. Has an improvement where the host side of containers veth pairs are directly created in the host namespace instead of moved there afterwards. Should help with the RTNL contention we are suffering from.

Fell into the dependency trap and had to bump some other stuff as well, like gomega and network-attachment-definition-client.

One specific issue was a circular dependency between hcsshim and containerd that required a replacement rule
`github.com/Microsoft/hcsshim => github.com/Microsoft/hcsshim v0.8.20` to the version pulled from containernetworking.

Also used a replacement repository for the netlink patch we had in place.
`github.com/vishvananda/netlink => github.com/jcaamano/netlink v1.1.1-0.20220831114501-3a761ed61db6`
Otherwise vendoring always requires some manual tinkering.
Advise if a different placement for this repo is more appropriate.

This PR needs to be squashed.

### CODE REVIEW

**Review by trozet — APPROVED** [review]

File: go-controller/pkg/cni/helper_linux.go [hunk 1]
```diff
@@ -428,40 +423,28 @@ func (pr *PodRequest) ConfigureInterface(podLister corev1listers.PodLister, kcli
...
-	if haveV6 {
-		err = netns.Do(func(hostNS ns.NetNS) error {
-			// deny IPv6 neighbor solicitations
-			dadSysctlIface := fmt.Sprintf("/proc/sys/net/ipv6/conf/%s/dad_transmits", contIface.Name)
-			if _, err := os.Stat(dadSysctlIface); !os.IsNotExist(err) {
-				err = setSysctl(dadSysctlIface, 0)
-				if err != nil {
-					klog.Warningf("Failed to disable IPv6 DAD: %q", err)
-				}
+	err = netns.Do(func(hostNS ns.NetNS) error {
```
Comment (dcbw) [inline go-controller/pkg/cni/helper_linux.go:426]: @jcaamano Looks like this reverts the "don't settle v4 addresses" bits?

Comment (jcaamano) [reply go-controller/pkg/cni/helper_linux.go]: Yup, fixed this.

**Comment by amorenoz** [comment]
Hello @jcaamano , @lmilleri and I were troubleshooting a problem with #2664 and it seems related to this PR.
In OVS hwol we were relying on `helper_linux.go`'s `setupNetwork` to configure the MAC address on the netdev but now it's not being done. Was this change intended?

**Comment by jcaamano** [comment]
> setupNetwork

@amorenoz
Hmm by the looks of it, that was an oversight on my side. The testing coverage did not help. 
Fixing in https://github.com/ovn-org/ovn-kubernetes/pull/3186
Could you give it a look?
cc @dcbw

**Comment by amorenoz** [comment]
Thanks for jumping on this @jcaamano !!

**Pull request merged**